components/threading/global_threads/thread_pane/thread_pane.tsx

```diff
@@ -53,9 +54,15 @@ const ThreadPane = ({
 
     const channel = useSelector((state: GlobalState) => getChannel(state, {id: channelId}));
     const post = useSelector((state: GlobalState) => getPost(state, thread.id));
-
+    const postsInThread = useSelector((state: GlobalState) => getPostsForThread(state, {rootId: post.id}));
     const selectHandler = useCallback(() => select(), []);
+    let unreadTimestamp = post.edit_at || post.create_at;
 
+    // if we have the whole thread, get the posts in it, sorted. First post - root post, second post - oldest reply. Use that timestamp
+    if (postsInThread.length > 1) {
+        const p = postsInThread[postsInThread.length - 2];
```

Comment:
Why do we do `postsInThread.length - 2` here? Isn't the oldest reply: `postsInThread.length-1`?

Reply:
@ashishbhate nope. the `postsInThread.length-1` item is the root post, see the comment above :)

Reply:
`postsInThread` is ordered from newest post  to the root post?

The comment is a bit confusing. `First post - root post`: From the word "first", I would assume it means index 0 in the array, so the opposite ordering of what it actually is.

Reply:
@ashishbhate yeah, reverse order. i'll fix the comment